### DISCUSSION TITLE
Advanced Thread Group Not working

### DISCUSSION THREAD

**Discussion by zetendra** [abstracta/jmeter-java-dsl#71] [Q&A] [answered]
System.out.println(testPlan(threadGroup()
                .rampTo(200, Duration.ofSeconds(10))
                .children(
                        httpSampler("http://my.service")
                )).run().overall().errorsCount());

Gives me error count as zero, can some one help me on this ?

**Comment by kirillyu** [comment]
You sure that it should have the different value? You can easy check it by using resultTreeVisualizer

**Reply by zetendra** [reply]
Thank you for responding, nothing is showing up in resultTreeVisualizer

Same worked like this

System.out.println(testPlan(threadGroup()
.rampToAndHold(200, Duration.ofSeconds(10), Duration.ofMillis(1))
.children(
httpSampler("http://my.service")
)).run().overall().errorsCount());


It worked when use rampToAndHold method

**Reply by rabelenda** [reply]
First: thank you for trying the DSL and reporting the issue! This is really helpful and helps us to constantly improve the DSL.

Yes, this looks like a bug. 

For the time being, for simple scenarios which only include a rampTime, use a ramp + hold with same duration in both, otherwise DSL will generate a thread group with ramp time but 0 iterations (resulting in no execution).

We will fix this in future version, we will let you know when the fix is released.

Regards

**Reply by zetendra** [reply]
Thank you @rabelenda

**Reply by rabelenda** [reply]
Hello, yesterday we released a [new version](https://github.com/abstracta/jmeter-java-dsl/releases/tag/v0.44) of the DSL which should fix the reported issue. 

Please let us know after you try it.

**Comment by zetendra** [comment, marked as answer]
Its perfect now, Working like a charm now with rampTo 👍 

INFO  Summariser:326 -  =    100 in 00:00:04 =   26.3/s Avg:  2321 Min:  1945 Max:  3062 Err:     0 (0.00%)
 =    100 in 00:00:04 =   26.3/s Avg:  2321 Min:  1945 Max:  3062 Err:     0 (0.00%)


Thank you so much @rabelenda
